radix-engine-tests/tests/blueprints/proof_creation/src/lib.rs

```diff
@@ -0,0 +1,274 @@
+            assert_eq!(proof.amount(), dec!(1));
+            proof.drop();
+            LocalAuthZone::clear();
+            bucket.burn();
+        }
+        pub fn create_proof_from_non_fungible_auth_zone_of_amount() {
+            let bucket = Self::prepare_non_fungible_proof();
+            let proof = LocalAuthZone::create_proof_of_amount(2, bucket.resource_address());
+            assert_eq!(proof.amount(), dec!(2));
+            proof.drop();
+            LocalAuthZone::clear();
+            bucket.burn();
+        }
+        pub fn create_proof_from_non_fungible_auth_zone_of_non_fungibles() {
+            let bucket = Self::prepare_non_fungible_proof();
+            let proof = LocalAuthZone::create_proof_of_non_fungibles(
+                btreeset!(
+                    NonFungibleLocalId::integer(1),
+                    NonFungibleLocalId::integer(2)
+                ),
+                bucket.resource_address(),
+            );
+            assert_eq!(proof.amount(), dec!(2));
+            proof.drop();
+            LocalAuthZone::clear();
+            bucket.burn();
+        }
+        pub fn create_proof_from_non_fungible_auth_zone_of_all() {
+            let bucket = Self::prepare_non_fungible_proof();
+            let proof = LocalAuthZone::create_proof_of_all(bucket.resource_address());
+            assert_eq!(proof.amount(), dec!(3));
+            proof.drop();
+            LocalAuthZone::clear();
+            bucket.burn();
+        }
+
+        //==================
+        // helper functions
+        //==================
+
+        pub fn create_fungible_bucket() -> Bucket {
+            ResourceBuilder::new_fungible()
+                .burnable(AccessRule::AllowAll, AccessRule::DenyAll)
+                .mint_initial_supply(100)
+        }
+
+        pub fn create_non_fungible_bucket() -> Bucket {
+            ResourceBuilder::new_integer_non_fungible()
+                .burnable(AccessRule::AllowAll, AccessRule::DenyAll)
+                .mint_initial_supply([
+                    (
+                        1u64.into(),
+                        DummyNFData {
+                            name: "NF One".to_owned(),
+                        },
+                    ),
+                    (
+                        2u64.into(),
+                        DummyNFData {
+                            name: "NF Two".to_owned(),
+                        },
+                    ),
+                    (
+                        3u64.into(),
+                        DummyNFData {
+                            name: "NF three".to_owned(),
+                        },
+                    ),
+                ])
+        }
+
+        pub fn create_fungible_vault() -> Vault {
+            Vault::with_bucket(Self::create_fungible_bucket())
+        }
+
+        pub fn create_non_fungible_vault() -> Vault {
+            Vault::with_bucket(Self::create_non_fungible_bucket())
+        }
+
+        pub fn prepare_fungible_proof() -> Bucket {
```

Comment:
Possibly `prepare_auth_zone_fungible_proof_to_bucket` would be clearer, I expected this function to return a proof.